src/core/integrators/brownian_inline.hpp

```diff
@@ -0,0 +1,571 @@
+  } else
+#endif
+  {
+    local_gamma = brownian.gamma;
+  }
+
+  bool aniso_flag; // particle anisotropy flag
+
+#ifdef PARTICLE_ANISOTROPY
+  // Particle frictional isotropy check.
+  aniso_flag =
+      (local_gamma[0] != local_gamma[1]) || (local_gamma[1] != local_gamma[2]);
+#endif
+
+  Utils::Vector3d force_body;
+  Utils::Vector3d delta_pos_body, delta_pos_lab;
+
+#ifdef PARTICLE_ANISOTROPY
+  if (aniso_flag) {
+    force_body = convert_vector_space_to_body(p, p.f.f);
+  }
+#endif
+
+  for (int j = 0; j < 3; j++) {
+    // Second (deterministic) term of the Eq. (14.39) of Schlick2010.
+    // Only a conservative part of the force is used here
+#ifdef PARTICLE_ANISOTROPY
+    if (aniso_flag) {
+      delta_pos_body[j] = force_body[j] * dt / (local_gamma[j]);
+    } else {
+#ifdef EXTERNAL_FORCES
+      if (!(p.p.ext_flag & COORD_FIXED(j)))
+#endif
+      {
+        p.r.p[j] += p.f.f[j] * dt / (local_gamma[j]);
+      }
+    }
+#else
+#ifdef EXTERNAL_FORCES
+    if (!(p.p.ext_flag & COORD_FIXED(j)))
+#endif
+    {
+      p.r.p[j] += p.f.f[j] * dt / (local_gamma);
+    }
+#endif // PARTICLE_ANISOTROPY
+  }
+#ifdef PARTICLE_ANISOTROPY
+  if (aniso_flag) {
+    delta_pos_lab = convert_vector_body_to_space(p, delta_pos_body);
+    for (int j = 0; j < 3; j++) {
+#ifdef EXTERNAL_FORCES
+      if (!(p.p.ext_flag & COORD_FIXED(j)))
+#endif
+      {
+        p.r.p[j] += delta_pos_lab[j];
+      }
+    }
+  }
+#endif // PARTICLE_ANISOTROPY
+}
+
+/** Set the terminal velocity driven by the conservative forces drag.
+ *  From eq. (14.34) in @cite Schlick2010.
+ *  @param[in,out] p              Particle
+ *  @param[in]     dt             Time interval
+ */
+inline void bd_drag_vel(Particle &p, double dt) {
+  // The friction tensor Z from the eq. (14.31) of Schlick2010:
+  Thermostat::GammaType local_gamma;
+
+#ifdef BROWNIAN_PER_PARTICLE
+  if (p.p.gamma >= Thermostat::GammaType{}) {
+    local_gamma = p.p.gamma;
+  } else
+#endif
+  {
+    local_gamma = brownian.gamma;
+  }
+
+  bool aniso_flag; // particle anisotropy flag
+
+#ifdef PARTICLE_ANISOTROPY
+  // Particle frictional isotropy check.
+  aniso_flag =
+      (local_gamma[0] != local_gamma[1]) || (local_gamma[1] != local_gamma[2]);
+#endif
+
+  Utils::Vector3d force_body;
+  Utils::Vector3d vel_body, vel_lab;
+
+#ifdef PARTICLE_ANISOTROPY
+  if (aniso_flag) {
+    force_body = convert_vector_space_to_body(p, p.f.f);
+  }
+#endif
+
+  for (int j = 0; j < 3; j++) {
+    // First (deterministic) term of the eq. (14.34) of Schlick2010 taking
+    // into account eq. (14.35). Only conservative part of the force is used
+    // here. NOTE: velocity is assigned here and propagated by thermal part
+    // further on top of it
+#ifdef PARTICLE_ANISOTROPY
+    if (aniso_flag) {
+      vel_body[j] = force_body[j] * dt / (local_gamma[j]);
+    } else {
+#ifdef EXTERNAL_FORCES
+      if (!(p.p.ext_flag & COORD_FIXED(j)))
+#endif
+      {
+        p.m.v[j] = p.f.f[j] / (local_gamma[j]);
```

Comment:
@psci2195 @RudolfWeeber Is this expression of the velocity intentionally missing the time step `dt`?

Reply:
@jngrad yes, the BD velocity does not depend on the time step. It is equal to the terminal viscous velocity.

Reply:
I still have a hard time understanding the code. When `PARTICLE_ANISOTROPY` is disabled, that function returns a velocity multiplied by the time step.

Reply:
thanks for catching! This is a mistake. The dt should be deleted there. The viscous test did not catch this because it covers the translationally isotropic particle only. I can fix this and increase a coverage.

Reply:
I've actually written C++ unit tests for Brownian and Langevin already. They caught that discrepancy, and I wasn't sure if it was intentional. Once the `dt` is removed on the line above and the `gamma_rotational` gets fixed in the other comment, my tests should pass. It would be nice to have an additional Python test to cover the rotational case, but we also shouldn't delay the merging of this PR for too long. I've already carried out a significant refactoring of the Brownian/Langevin thermostats thanks to @fweik to make them testable with Boost and non-global, but this PR is blocking me.